Mathlib/Order/Nucleus.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2024 Christian Krause. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chriara Cimino, Christian Krause
+-/
+import Mathlib.Order.CompleteBooleanAlgebra
+
+/-!
+# Nucleus
+Locales are the dual concept to Frames.
+A Nucleus is a function between Frames which corresponds to a sublocale.
+https://ncatlab.org/nlab/show/nucleus
+-/
+variable {X : Type*} [Order.Frame X]
```

Comment:
Do you need `Order.Frame` here? Why not just `SemilatticeInf`?

Reply:
What about the lemmas where I need other Frame properties (like \top and suprema)?